allennlp/training/callbacks/callback_handler.py

```diff
@@ -0,0 +1,33 @@
+"""
+blah
+"""
+from typing import Iterable, TypeVar, Generic
+
+from allennlp.common.registrable import Registrable
+from allennlp.training.callbacks.callback import Callback
+
+State = TypeVar('State')  # pylint: disable=invalid-name
+
+class CallbackHandler(Registrable, Generic[State]):
```

Comment:
The entire need for generics in this class also goes away if you just use `Trainer`.